server/session.py

```diff
@@ -109,7 +109,7 @@ def run():
 		return
 
 	if state == STATE_LISTEN:
-		if not ser.dsr:
+		if not get_hshk:
```

Comment:
I think these all need to be "get_hshk()" i.e. add brackets to make it a function call - it could explain the related issue you raised

Reply:
Good spot, this was it. Thanks